src/components/Hoverable/ActiveHoverable.tsx

```diff
@@ -0,0 +1,143 @@
+
+        const scrollingListener = DeviceEventEmitter.addListener(CONST.EVENTS.SCROLLING, (scrolling) => {
+            isScrolling.current = scrolling;
+            if (!isScrolling.current) {
+                setIsHovered(isHoveredRef.current);
+            }
+        });
+
+        return () => scrollingListener.remove();
+    }, [shouldHandleScroll]);
+
+    useEffect(() => {
+        // Do not mount a listener if the component is not hovered
+        if (!isHovered) {
+            return;
+        }
+        /**
+         * Checks the hover state of a component and updates it based on the event target.
+         * This is necessary to handle cases where the hover state might get stuck due to an unreliable mouseleave trigger,
+         * such as when an element is removed before the mouseleave event is triggered.
+         * @param event The hover event object.
+         */
+        const unsetHoveredIfOutside = (event: MouseEvent) => {
+            if (!ref.current || ref.current.contains(event.target as Node)) {
+                return;
+            }
+
+            setIsHovered(false);
+        };
+
+        document.addEventListener('mouseover', unsetHoveredIfOutside);
+
+        return () => document.removeEventListener('mouseover', unsetHoveredIfOutside);
+    }, [isHovered, ref]);
+
+    useEffect(() => {
+        const unsetHoveredWhenDocumentIsHidden = () => document.visibilityState === 'hidden' && setIsHovered(false);
+
+        document.addEventListener('visibilitychange', unsetHoveredWhenDocumentIsHidden);
+
+        return () => document.removeEventListener('visibilitychange', unsetHoveredWhenDocumentIsHidden);
+    }, []);
+
+    const child = useMemo(() => (typeof children === 'function' ? children(!isScrolling.current && isHovered) : children), [children, isHovered]);
+
+    const onMouseEnter = useCallback(
+        (e: MouseEvent) => {
+            updateIsHovered(true);
+            child.props.onMouseEnter?.(e);
+        },
+        [updateIsHovered, child.props],
+    );
+
+    const onMouseLeave = useCallback(
+        (e: MouseEvent) => {
+            updateIsHovered(false);
+            child.props.onMouseLeave?.(e);
+        },
+        [updateIsHovered, child.props],
+    );
+
+    const disableHoveredOnBlur = useCallback(
+        (event: MouseEvent) => {
+            // Check if the blur event occurred due to clicking outside the element
+            // and the wrapperView contains the element that caused the blur and reset isHovered
+            if (!ref.current?.contains(event.target as Node) && !ref.current?.contains(event.relatedTarget as Node)) {
+                setIsHovered(false);
+            }
+
+            child.props.onBlur?.(event);
+        },
+        [child.props],
+    );
+
+    // We need to access the ref of a children from both parent and current component
+    // So we pass it to current ref and assign it once again to the child ref prop
+    const hijackRef = (el: HTMLElement) => {
```

Comment:
Could this be extracted to a hook like...

```ts

class HijackingRefObject<T> implements MutableRefObject<T> {
  private _targetRef: Ref<T>;
  private _current: T | null = null;

  constructor(targetRef: Ref<T>) {
    this._targetRef = targetRef;
  }

  get current(): T | null {
    return this._current;
  }

  set current(value: T | null) {
    this._current = value;
    assignRef(targetRef, value);
  }
}

function useHijackingRef<T>(targetRef: Ref<T>): MutableRefObject<T> {
  return new HijackingRefObject<T>(targetRef);
}

```

Reply:
I remember we have other places where we "hijack" so it would be cool to be able to reuse this. But I couldn't think of a simpler thing than this class with a custom setter.

Reply:
Thanks for the idea! How would you use it in the `Hoverable` case? We need to 'hijack' the ref directly in the `cloneElement`, which kind of defeats the usage of a hook. Please correct me if my thinking here is wrong 👍

Reply:
I thought it would be used like...

```ts
const elementRef = useHijackingRef(child.ref);
```

But my suggested code is much longer than the current local function, and I have some second thoughts here. Maybe we can leave this as an inspiration for the future, in case we work on another piece of code that does "ref hijacking".